ee/clickhouse/queries/paths/path_event_query.py

```diff
@@ -116,3 +117,29 @@ def _get_event_query(self) -> Tuple[str, Dict[str, Any]]:
             return f" AND {' AND '.join(conditions)}", params
 
         return "", {}
+
+    def _should_query_url(self) -> bool:
+        if self._filter.target_events == [] and self._filter.custom_events == []:
+            if PAGEVIEW_EVENT not in self._filter.exclude_events:
+                return True
+        else:
+            if self._filter.include_pageviews:
+                return True
+
+        return False
+
+    def _should_query_screen(self) -> bool:
+        if self._filter.target_events == [] and self._filter.custom_events == []:
+            if SCREEN_EVENT not in self._filter.exclude_events:
```

Comment:
The nested if is playing with my head here. Is this intentional, why this way??

Reply:
A relic of the path I walked to get here. Refactoring.

I'm a non-Markov process 😂